server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -472,6 +472,26 @@ public long sizeInBytesByMinGen(long minGeneration) {
         }
     }
 
+    /**
+     * Returns the age of the oldest translog file with at least the given generation
+     */
+    public long getOldestTranslogAgeInMillisByMinGen(long minGeneration) {
```

Comment:
Perhaps delegate `earliestLastModifiedAge` to `getOldestTranslogAgeInMillisByMinGen(-1L)`?